Android/app/src/main/java/app/intra/LocalhostResolver.java

```diff
@@ -42,37 +44,37 @@
   private final DatagramSocket socket;
   private final DnsVpnService vpnService;
 
-  @Nullable
-  ServerConnection serverConnection;
-
   /**
    * Get a localhost resolver, bound to a port and ready to be started by calling start().
    * @param vpnService The VPN service that owns this server.
-   * @param serverConnection The connection to an upstream DOH server.
    * @return The localhost resolver, or null if it couldn't be started.
    */
-  public static LocalhostResolver get(DnsVpnService vpnService, ServerConnection serverConnection) {
+  public static LocalhostResolver get(DnsVpnService vpnService) {
     DatagramSocket socket = null;
     try {
-      SocketAddress bindaddr = new InetSocketAddress("localhost", 0);
+      InetAddress localhost = InetAddress.getByAddress(new byte[]{127, 0, 0, 1});
```

Comment:
Can you use `InetAddress.getLocalHost()`?

Reply:
`getLocalHost()` seems scary.  The [docs](https://developer.android.com/reference/java/net/InetAddress#getLocalHost()) don't promise that it actually resolves to a "loopback" IP; they just say:

> Returns the address of the local host. This is achieved by retrieving the name of the host from the system, then resolving that name into an InetAddress.

[`getLoopbackAddress`](https://developer.android.com/reference/java/net/InetAddress#getLoopbackAddress()) actually does promise to do what we want, but on my phone it returns IPv6 localhost, which doesn't seem to work with tun2socks right now.